ext/dom/tests/modern/common/Element_getElementsByClassName.phpt

```diff
@@ -0,0 +1,42 @@
+--TEST--
+Dom\Element::getElementsByClassName()
+--EXTENSIONS--
+dom
+--FILE--
+<?php
+
+$dom = Dom\HTMLDocument::createFromString(<<<HTML
+<div class="  foo bar ">
+    <p class="bar">
+        <p class="bar" name="here">1</p>
+        <p name="here">2</p>
+        <p name="here">3</p>
+        <p class="bar" name="here">4</p>
+    </p>
+    <b class="foo bars"></b>
+</div>
+HTML, LIBXML_NOERROR);
```

Comment:
I might be misunderstanding the spec or not seeing how your patch achieves the desired results.
But it seems to me that in quirk mode if you have HTML with a class written as `<div class="mYcuStoMnAme">` you should be able to query it by doing `$dom->documentElement->getElementsByClassName("myCustomName");`

In any case can you add some quirk mode tests just to confirm?

Reply:
Yeah the tests were insufficient. I now split the test in a quirks and non quirks one.
You're correct about your understanding of the spec.

The patch achieves this behaviour by ASCII lowercasing the string when parsing the set, and when doing a lookup, if in quirks mode.

Reply:
Ah right I missed the insertion into the HT was lowercased too!